### PR TITLE
Add `torch.where` overload with only condition argument

### PR DESCRIPTION
**Pull request by umanwizard**
Requested in #21798

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -5839,7 +5839,7 @@ def merge_dicts(*dicts):
 
 add_docstr(torch.where,
            r"""
-where(condition, x, y) -> Tensor
+.. function:: where(condition, x, y) -> Tensor
```
Comment (gchanan) [inline torch/_torch_docs.py]: isn't this either a different function or the optional `x, y` need to be explained in some way?

Comment (gchanan) [reply torch/_torch_docs.py]: whoops, didn't see the line below, my mistake.

**Review by gchanan — APPROVED** [review]

**Comment by apaszke** [comment]
Wow, NumPy API is so weird.

**Comment by zhangguanheng66** [comment]
@umanwizard Could you rebase and resolve the merge conflicts? I tried to land the diff and got merge conflict warnings. Thanks.

**Pull request closed without merging**